src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -159,7 +160,7 @@ case class Task(taskId: Task.Id, runSpecVersion: Timestamp, status: Task.Status)
           if (status.hasUnreachableTime) status.getUnreachableTime
           else Timestamp.fromTaskStatus(status)
 
-        since.expired(now, by = timeout)
+        since.expired(now, by = timeout) || timeout == 0.seconds
```

Comment:
I would expect that since.expired should handle case of timeout 0 seconds by itself. Why we have to handle it separately?

Reply:
`expired` checks `now + by > since` but not `now + by >= since`.

Reply:
so shouldn't we patch the expired method? Since Timestamp is our class

Reply:
We could. However, I don't think one would expect `now.expired(now, by = 0.seconds)` to be `true`. Or maybe it is.